tests/e2escenarios/core_beta_flow.go

```diff
@@ -0,0 +1,169 @@
+			helper.FileShouldContainSubstring(configFile, "Application: myapp")
+			helper.FileShouldContainSubstring(configFile, "SourceType: local")
+			// SourcePath should be relative
+			//helper.FileShouldContainSubstring(configFile, "SourceLocation: .")
+			helper.FileShouldContainSubstring(configFile, "Project: "+project)
+
+			configSession := helper.CmdShouldPass(odo, append([]string{"config", "set", "--env", "FOO=bar"}, extraArgs...)...)
+			// output of the commands should point user to running "odo push"
+			// currently failing
+			Eventually(configSession).Should(gbytes.Say("odo push"))
+			helper.FileShouldContainSubstring(configFile, "Name: FOO")
+			helper.FileShouldContainSubstring(configFile, "Value: bar")
+
+			urlCreateSession := helper.CmdShouldPass(odo, append([]string{"url", "create", "--port", "8080"}, extraArgs...)...)
+			// output of the commands should point user to running "odo push"
+			Eventually(urlCreateSession).Should(gbytes.Say("odo push"))
+			helper.FileShouldContainSubstring(configFile, "Url:")
+			helper.FileShouldContainSubstring(configFile, "Port: 8080")
+
+			helper.CmdShouldPass(odo, append([]string{"push"}, extraArgs...)...)
+
+			dcSession := oc.GetComponentDC("mycomponent", "myapp", project)
+			Eventually(dcSession).Should(gbytes.Say("app.kubernetes.io/component-name: mycomponent"))
+			Eventually(dcSession).Should(gbytes.Say("app.kubernetes.io/component-source-type: local"))
+			Eventually(dcSession).Should(gbytes.Say("app.kubernetes.io/component-type: java"))
+			Eventually(dcSession).Should(gbytes.Say("app.kubernetes.io/name: myapp"))
+			Eventually(dcSession).Should(gbytes.Say("name: mycomponent-myapp"))
+			// DC should have env variable
+			// currently failing
+			Eventually(dcSession).Should(gbytes.Say("name: FOO"))
+			Eventually(dcSession).Should(gbytes.Say("value: bar"))
+
+			routeSession := oc.GetComponentRoutes("mycomponent", "myapp", project)
+			// check that route is pointing gto right port and component
+			Eventually(routeSession).Should(gbytes.Say("targetPort: 8080"))
+			Eventually(routeSession).Should(gbytes.Say("name: mycomponent-myapp"))
+			url := oc.GetFirstURL("mycomponent", "myapp", project)
+
+			helper.HttpWaitFor("http://"+url, "Hello World from Javalin!", 10, 5)
+		}
+
+		Context("when component is in the current directory", func() {
+
+			// we will be testing components that are created from the current directory
+			// switch to the clean context dir before each test
+			JustBeforeEach(func() {
+				originalDir = helper.Getwd()
+				helper.Chdir(context)
+			})
+
+			// go back to original directory after each test
+			JustAfterEach(func() {
+				helper.Chdir(originalDir)
+			})
+
+			It("'odo component' should fail if there already is .odo dir", func() {
+				helper.CmdShouldPass("odo", "component", "create", "nodejs")
+				helper.CmdShouldFail("odo", "component", "create", "nodejs")
+			})
+
+			It("'odo config' should fail if there is no .odo dir", func() {
```

Comment:
this should be put before previous test

Reply:
ideally command will run successfully because previous test will create `.odo`

Reply:
Each `It` is an independent test case. There can't be any dependencies on the order in which the `It` sections are run.